src/plugins/intel_gpu/src/runtime/device.cpp

```diff
@@ -4,64 +4,128 @@
+                return it->second;
+            }
+        }
+
+        return get_ops_for_mad(dt);
+    }
+
+    float get_ops_for_mad(data_types dt) const {
+        auto it = ops_mad.find(dt);
+        if (it != ops_mad.end()) {
+            return it->second;
         } else {
-            // fma * simd size
-            opsPerComputeBlock = 2 * 16;
+            return 0;
         }
-        break;
     }
-    case cldnn::data_types::f32: {
-        // fma * simd size
-        opsPerComputeBlock = 2 * 8;
-        break;
+};
+
+const uint8_t MAX_REVISION = 0xff;
+
+const std::vector<DeviceOps> device_ops_table = {
+//    | platform | device_id | gfx_ver                                 | MAD                   | DPAS (imad)           | DP4A (immad)  |
+//    |          |           |                                         |  fp64 |  fp32 |  fp16 |  fp16 |  int8 |  int4 |  fp16 |  int8 |
+    { "Legacy",   {},         { { 0,  0,  0}, {12,  9, MAX_REVISION} }, {     0,     16,     32,      0,      0,      0,      0,      0 } },    // TGL, ADL-S
+    { "DG1",      {},         { {12, 10,  0}                         }, {     0,     16,     32,      0,      0,      0,      0,      0 } },
+    { "DG2",      {},         { {12, 55,  0}, {12, 57, MAX_REVISION} }, {     0,     16,     32,      0,      0,      0,    128,    256 } },
+    { "PVC_XL",   {},         { {12, 60,  0}, {12, 60, 1}            }, {    16,     32,     64,      0,      0,      0,    512,   1024 } },
+    { "PVC_XT",   {},         { {12, 60,  3}, {12, 61, 7}            }, {    32,     32,     64,      0,      0,      0,    512,   1024 } },
+    { "MTL/ARL",  {},         { {12, 70,  0}, {12, 74, MAX_REVISION} }, {   0.5,     16,     32,    128,    256,      0,      0,     64 } },
+    { "ATS",      { 0x020A, 0x0210 },
+                              {},                                       {     0,     16,     32,    128,    256,    512,      0,      0 } },
+    { "BMG",      {},         { {20,  1,  0}, {20,  2, MAX_REVISION} }, {     1,     16,     32,    128,    256,    512,      0,      0 } },
+    { "LNL/PTL",  {},         { {20,  4,  0}, {30,  1, MAX_REVISION} }, {     1,     16,     32,    128,    256,    512,      0,      0 } },
+};
+
+float device::get_gops(data_types dt) const {
+    // WA: The u8 type isn't accounted for in the device_ops_table, since it's the same as i8.
+    dt = dt == data_types::u8 ? data_types::i8 : dt;
+
+    auto info = get_info();
+    if (info.vendor_id != INTEL_VENDOR_ID) {
+        // GOPS calculation is not supported for non Intel GPUs
+        return 0.0f;
     }
 
-    default: OPENVINO_ASSERT(false, "[GPU] get_gops: unsupported precision: ", dt);
+    auto freqGHz = info.gpu_frequency / 1000.f;
+    auto numEUs = info.execution_units_count * (info.arch < gpu_arch::xe2 ? 1 : 2);
+    auto opsPerEU = 0.f;
+
+    auto it = std::find_if(device_ops_table.begin(), device_ops_table.end(),
+        [&info](auto& ops) {
```

Comment:
nit: `ops` -> `entry`. This is comparing against `entry in ops table`. So `ops` is a misnomer

Reply:
apply it